src/consume.ts

```diff
@@ -656,18 +658,17 @@ function messageViewerStartPollingCommand(
           serialized.value.includes(index),
         );
 
-        // i want to drop the comment in favor of filename and possibly do a preview tab
-        workspace
-          .openTextDocument({
-            content: `// message ${message.key} from ${topic.name}\n${JSON.stringify(payload, null, 2)}`,
-            language: "jsonc",
+        // read-only document buffer with the message content
+        const filename = `${topic.name}-message-${index}.json`;
+        const uri: Uri = new MessageDocumentProvider().resourceToUri(payload, filename);
```

Comment:
Is there a limit to how long the URI can be? A message may have `value` as long as 1MB.

Reply:
Good point - I'm not sure of any URI size limits, but I double-checked against our `stress_test_json` topic, and it looks like those big messages are still loading fine:
<img width="2302" alt="image" src="https://github.com/user-attachments/assets/7d03f036-2f88-485e-9250-a64dc767a2d7" />